src/views/home/components/list/list.view.tsx

```diff
@@ -0,0 +1,86 @@
+import { FC, useState } from "react";
+
+import useListStyles from "src/views/home/components/list/list.styles";
+import Card from "src/views/shared/card/card.view";
+import { ReactComponent as SearchIcon } from "src/assets/icons/search.svg";
+import Typography from "src/views/shared/typography/typography.view";
+import TokenIcon from "src/views/shared/token-icon/token-icon.view";
+import { Chain, Token } from "src/domain";
+
+interface ListProps {
+  placeholder: string;
+  list:
+    | { type: "chain"; items: Chain[]; onClick: (chain: Chain) => void }
+    | { type: "token"; items: Token[]; onClick: (token: Token) => void };
+}
+
+const List: FC<ListProps> = ({ placeholder, list }) => {
+  const classes = useListStyles();
+  const [values, setValues] = useState(list);
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const filterChain = (chain: Chain) =>
```

Comment:
I would put this `filter` functions outside of this function to avoid creating them each time that you call the function.